test/unit/sync_mediator_test.rb

```diff
@@ -118,18 +121,31 @@ def test_calling_instance_sync_does_not_allow_instances_to_affect_the_global_cac
     bad_citizen = BadCitizen.new('foo')
     stub_kubectl_response('get', 'BadCitizen', *@params, resp: { "items" => [bad_citizen.kubectl_response] }, times: 1)
 
-    mediator.sync([bad_citizen])
+    # Cache is only warmed if we batch fetch
+    mediator.sync([bad_citizen] * (KubernetesDeploy::SyncMediator::LARGE_BATCH_THRESHOLD + 1))
     assert_equal bad_citizen.kubectl_response, mediator.get_instance('BadCitizen', bad_citizen.name) # still cached
   end
 
   def test_sync_calls_sync_on_each_instance
-    stub_kubectl_response('get', 'FakeDeployment', *@params,
-      resp: { "items" => [@fake_deployment.kubectl_response] }, times: 1)
-    stub_kubectl_response('get', 'FakePod', *@params, resp: { "items" => [@fake_pod.kubectl_response] }, times: 1)
-    all_cm = [@fake_cm.kubectl_response, @fake_cm2.kubectl_response]
-    stub_kubectl_response('get', 'FakeConfigMap', *@params, resp: { "items" => all_cm }, times: 1)
+    test_resources = [@fake_pod, @fake_cm, @fake_deployment]
+    test_resources.each { |r| r.expects(:sync).once }
+    mediator.sync(test_resources)
+  end
+
+  def test_sync_does_not_batch_with_few_resources
+    stub_kubectl_response('get', 'FakeConfigMap', @fake_cm.name, *@params,
+      resp: { "items" => [@fake_cm.kubectl_response] }, times: 0)
+
+    test_resources = [@fake_cm]
+    test_resources.each { |r| r.expects(:sync).once }
+    mediator.sync(test_resources)
+  end
+
+  def test_sync_does_batch_with_enough_resources
```

Comment:
What value does this add beyond `test_sync_caches_the_types_of_the_dependencies_of_the_instances_given` and `test_sync_caches_the_types_of_the_instances_given` (which should probably have their names amended)? Because this test is stubbing the instance syncs, it isn't actually proving anything about the cache, just that the list api call was made (which those existing tests also do).